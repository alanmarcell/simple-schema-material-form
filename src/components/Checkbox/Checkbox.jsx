import React from 'react';
import { merge } from 'ramda';
import MaterialCheckbox from 'material-ui/Checkbox';
import { FormControlLabel, FormGroup } from 'material-ui/Form';
import SelectionControl from '../SelectionControl';

const Control = ({
  setDoc, doc, fieldName, i,
}) => {
  return (<MaterialCheckbox
    onChange={(_event, val) => setDoc(merge(doc, { [fieldName]: { ...doc[fieldName], [i.key]: val } }))}
  />);
};

const Checkbox = props => {
  const {
    setDoc, label, selectOptions = [], fieldName, doc,
  } = props;

  return (
    <SelectionControl
      label={label}
    >
      <FormGroup row>
        {selectOptions.map(i =>
          (<FormControlLabel
            key={i.key}
            control={<Control {...{
              i, doc, fieldName, setDoc,
            }}
            />}
            label={i.label}
          />))
        }
      </FormGroup>
    </SelectionControl>
  );
};

export default Checkbox;
