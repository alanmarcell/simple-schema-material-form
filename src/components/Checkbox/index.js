import CheckboxComponent from './Checkbox';

export default CheckboxComponent;
