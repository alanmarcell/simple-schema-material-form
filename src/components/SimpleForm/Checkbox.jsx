import React from 'react';
import { merge } from 'ramda';
import MaterialCheckbox from 'material-ui/Checkbox';
import { FormControl, FormLabel, FormControlLabel, FormGroup } from 'material-ui/Form';

const Checkbox = props => {
  const {
    setDoc, label, selectOptions = [], fieldName, doc,
  } = props;

  return (
    <FormControl >
      <FormLabel>{label}</FormLabel>
      <FormGroup row>
        {selectOptions.map(i =>
          (<FormControlLabel
            key={i.key}
            control={<MaterialCheckbox
              checked={i.checked}
              onChange={(_event, val) => setDoc(merge(doc, { [fieldName]: { ...doc[fieldName], [i.key]: val } }))}
              value={i.key || ''}
              id={i.key}
            />}
            label={i.label}
          />))
        }
      </FormGroup>
    </FormControl >
  );
};

// const Checkbox = props => {
//   const inputProps = {
//     ...props,
//   };
//   return <CheckboxField selectOptions={[{ label: 1, key: 'um' }, { label: 2, key: 'dois' }]} {...inputProps} />;
// };

export default Checkbox;
