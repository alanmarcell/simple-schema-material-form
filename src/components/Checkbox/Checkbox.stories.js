import React from 'react';
import { storiesOf } from '@storybook/react';
import Checkbox from './';
import withTests from '../../stories/withTests';

storiesOf('Fields/Checkbox', module)
  .addDecorator(withTests('Checkbox'))
  .add('simple', () => (
    <Checkbox selectOptions={[{ label: 1, key: 'um' }, { label: 2, key: 'dois' }]} fieldName="test" />
  ));
