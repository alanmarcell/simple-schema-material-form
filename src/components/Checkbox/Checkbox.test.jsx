import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Checkbox from './Checkbox';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Checkbox fieldName="test" />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow a Checkbox', () => {
  const setDoc = jest.fn();
  const CheckboxComponent = (<Checkbox
    fieldName="test"
    setDoc={setDoc}
    doc={{}}
    selectOptions={[{ label: 1, key: 'um' }, { label: 2, key: 'dois' }]}
  />);

  const component = shallow(CheckboxComponent);
  const { control } = component.find({ label: 1 }).props();
  const Control = shallow(control);

  Control.simulate('change', null, true);
  Control.simulate('change', null, false);

  // Inspect doc prop field key
  expect(setDoc).toBeCalledWith({ test: { um: true } });
  expect(setDoc).toHaveBeenLastCalledWith({ test: { um: false } });
  expect(setDoc).toHaveBeenCalled();
  expect(component).toMatchSnapshot();
});

