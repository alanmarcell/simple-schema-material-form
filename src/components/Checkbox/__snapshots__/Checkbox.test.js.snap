// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow a Checkbox 1`] = `
<WithStyles(FormControl)>
  <WithStyles(FormLabel) />
  <WithStyles(FormGroup)
    row={true}
  >
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Checkbox)
          checked={undefined}
          id="um"
          onChange={[Function]}
          value="um"
        />
      }
      key="um"
      label={1}
    />
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Checkbox)
          checked={undefined}
          id="dois"
          onChange={[Function]}
          value="dois"
        />
      }
      key="dois"
      label={2}
    />
  </WithStyles(FormGroup)>
</WithStyles(FormControl)>
`;
